src/main/rx/cc2500_redpine.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+#include <sys/_stdint.h>
+
+#include "platform.h"
+
+#ifdef USE_RX_REDPINE_SPI
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "pg/rx.h"
+#include "pg/rx_spi.h"
+
+#include "common/maths.h"
+#include "common/utils.h"
+
+#include "config/feature.h"
+
+#include "drivers/adc.h"
+#include "drivers/rx/rx_spi.h"
+#include "drivers/rx/rx_cc2500.h"
+#include "drivers/io.h"
+#include "drivers/io_def.h"
+#include "drivers/io_types.h"
+#include "drivers/resource.h"
+#include "drivers/system.h"
+#include "drivers/time.h"
+
+#include "fc/config.h"
+
+#include "rx/rx_spi_common.h"
+#include "rx/cc2500_common.h"
+#include "rx/cc2500_frsky_common.h"
+#include "rx/cc2500_frsky_shared.h"
+
+#include "cc2500_redpine.h"
+
+extern const uint16_t crcTable[];
+
+
+#define TOTAL_PACKET_TIME 1500
+#define CHANNEL_START 3
+void redpineSetRcData(uint16_t *rcData, const uint8_t *packet)
+{
+    uint16_t channelValue;
+
+    //4 stick channels (11-bit)
+    channelValue = (uint16_t)((packet[CHANNEL_START+1] << 8) & 0x700) | packet[CHANNEL_START];
+    rcData[0] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+2] << 4) & 0x7F0) | ((packet[CHANNEL_START+1] >> 4) & 0xF);
+    rcData[1] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+4] << 8) & 0x700) | packet[CHANNEL_START+3];
+    rcData[2] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+5] << 4) & 0x7F0) | ((packet[CHANNEL_START+4] >> 4) & 0xF);
+    rcData[3] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;    
+    //12 1-bit aux channels - first 4 are interleaved with stick channels
+    rcData[4]= (packet[CHANNEL_START + 1] & 0x08) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
```

Comment:
Would be nice to have some choice here with respect to having extra 'analogue' channels (for adjrange settings), or channels to map 6 position switches. How about having the TX send a 'channel map' id as part of the binding procedure, and the RX then process data accordingly?

Reply:
Yeah that would be a great feature to add.  I was trying to minimize changes to other code and creating a small packet size.  

For the 6 positions switch example, they could each be sent as separate 1-bit channels, that is currently what I do for 3 position switches, but it would be easier to set up as an analog channel.

Maybe the option to have the option of 4, 8, or 16 11-bit analog channels?  I could make it completely flexible to select any number of analog or 11-bit channels but that seems excessive.

Reply:
Agreed, there is value in simplicity. I think being able to send additional analogue channels for adjustment ranges is probably the most valuable of the 'extra options'. But for now starting out with the fixed allocation is fine.